client/dashboard/sites/settings-site-visibility/launch-form.tsx

```diff
@@ -1,12 +1,91 @@
+import { formatCurrency } from '@automattic/number-formatters';
+import { useQuery } from '@tanstack/react-query';
 import {
+	__experimentalHStack as HStack,
 	__experimentalText as Text,
 	__experimentalVStack as VStack,
 	Button,
+	ExternalLink,
 } from '@wordpress/components';
-import { __ } from '@wordpress/i18n';
+import { createInterpolateElement } from '@wordpress/element';
+import { __, _n, sprintf } from '@wordpress/i18n';
 import { addQueryArgs } from '@wordpress/url';
+import { agencyBlogQuery } from '../../app/queries';
 import type { Site } from '../../data/types';
 
+function useAgencyBillingMessage( site: Site ) {
+	const { data, isLoading: isLoading, isError: isError } = useQuery( agencyBlogQuery( site.ID ) );
+	if ( ! data ) {
+		return undefined;
+	}
+
+	const priceInfoIsDefined =
+		Number.isFinite( data.prices?.actual_price ) && typeof data.prices?.currency === 'string';
+
+	if ( isLoading || isError || ! priceInfoIsDefined ) {
```

Comment:
This `isLoading` is curious. It's only `true` when we're loading the data for the first time. But if we're loading for the first time then `data` will be undefined, in which case we've already returned early on line 19 🤔